### ISSUE TITLE
wait_for_request function doesn't works correctly

### ISSUE THREAD

**Issue by StanleyHuang31** [wkeeling/selenium-wire#73]
Imagine that there'are few request with similar URL were received when you perform an action from Web page.
```
request_1, URL: https://192.168.1.1/redfish/v1
request_2, URL: https://192.168.1.1/redfish
```

And then I wanna capture request_2 through wait_for_request function, but I would get the request_1 due to the URL path is similar.

Do you think is it better to change the condition as following, so we could get the exactly request we want?
```
if match_url == indexed_request.path: 
```

https://github.com/wkeeling/selenium-wire/blob/a42fba54d7e3500e23e729e4ebed3fa4de6ea47a/seleniumwire/proxy/storage.py#L281

**Comment by wkeeling** [comment]
Thanks for raising this. You're right that the current matching behaviour would match both example URLs. The behaviour as you've discovered is pretty crude and just does a string contains.

I think going the full regex route might be the best option here. That way we could explicitly match the end of the URL using the `$`. For example:

`wait_for_request('https://192.168.1.1/redfish$')  # Would only match request_1`

I'll look at making that change in the next release.

**Comment by wkeeling** [comment]
This has been fixed in version 2.1.0. `driver.wait_for_request()` now accepts a regex.
